### PR TITLE
GH-2214: Fix Manual Nack with Zero Sleep

### PR DESCRIPTION
**Pull request by garyrussell**
Resolves https://github.com/spring-projects/spring-kafka/issues/2214

Regression; manual nack with zero sleep went into a tight loop since
2.8.4.

Add a copy of the current test, with a zero sleep; also verify that
nacking the first and last records work.

**cherry-pick to 2.9.x, 2.8.x**

### CODE REVIEW

File: spring-kafka/src/test/java/org/springframework/kafka/listener/ManualNackRecordZeroSleepTests.java [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+		}
+
+		@SuppressWarnings({ "rawtypes", "unchecked" })
+		@Bean
+		public ConcurrentKafkaListenerContainerFactory kafkaListenerContainerFactory() {
+			ConcurrentKafkaListenerContainerFactory factory = new ConcurrentKafkaListenerContainerFactory();
+			factory.setConsumerFactory(consumerFactory());
+			factory.getContainerProperties().setAckMode(AckMode.MANUAL);
+			factory.getContainerProperties().setMissingTopicsFatal(false);
+			factory.setRecordInterceptor(new RecordInterceptor() {
```
Comment (artembilan) [inline spring-kafka/src/test/java/org/springframework/kafka/listener/ManualNackRecordZeroSleepTests.java:249]: Do we really need this interceptor to be copied in this test scope?
I had a pain back-porting it yesterday for other test class: some API changes in Kafka client.

Thanks

Comment (garyrussell) [reply spring-kafka/src/test/java/org/springframework/kafka/listener/ManualNackRecordZeroSleepTests.java:249]: No; just a copy/paste artifact; will remove.

**Comment by sta-szek** [comment]
hi, i left question there: https://github.com/spring-projects/spring-kafka/issues/2214#issuecomment-1091407124 but i feel this is better place to ask
